src/pollux/utils/JWT.ts

```diff
@@ -18,7 +20,39 @@ export class JWT {
     this.castor = castor;
   }
 
-  private async resolve(did: string): Promise<didResolver.DIDResolutionResult> {
+  async verify(
+    options: {
+      issuerDID: DID,
+      holderDID?: DID,
+      jws: string
+    }
+  ): Promise<boolean> {
+    try {
+      const { issuerDID, jws, holderDID } = options;
+      const resolved = await this.resolve(issuerDID.toString());
+      const verificationMethod = resolved.didDocument?.verificationMethod;
+      if (!verificationMethod) {
+        throw new Error("Invalid did document");
+      }
+
+      const validVerificationMethod = didJWT.verifyJWS(jws, verificationMethod);
+      const jwtObject = JWTCredential.fromJWS(jws);
+
+      if (jwtObject.issuer !== issuerDID.toString()) {
+        throw new Error("Invalid issuer");
+      }
+
+      if (jwtObject.isCredential && holderDID && holderDID.toString() !== jwtObject.subject) {
+        throw new Error("Invalid subject (holder)");
```

Comment:
Replace with a custom error.

Reply:
This error is ignored and inside a try catc